dnf5daemon-server/package.cpp

```diff
@@ -196,3 +196,158 @@ dnfdaemon::KeyValueMap package_to_map(
     }
     return dbus_package;
 }
+
+void add_string_list(json_object * json_pkg, const char * cattr, const std::vector<std::string> & vector) {
+    json_object * array = json_object_new_array();
+    json_object_object_add(json_pkg, cattr, array);
+    for (const auto & elem : vector) {
+        json_object_array_add(array, json_object_new_string(elem.c_str()));
+    }
+}
+
+std::string package_to_json(
+    const libdnf5::rpm::Package & libdnf_package, [[maybe_unused]] const std::vector<std::string> & attributes) {
+    json_object * json_pkg = json_object_new_object();
+
+    // add package id by default
+    json_object_object_add(json_pkg, "id", json_object_new_int(libdnf_package.get_id().id));
+
+    // attributes required by client
+    for (const auto & attr : attributes) {
+        const auto * const cattr = attr.c_str();
+        auto it = package_attributes.find(attr);
+        if (it == package_attributes.end()) {
+            throw std::runtime_error(fmt::format("Package attribute '{}' not supported", attr));
+        }
+        switch (it->second) {
+            case PackageAttribute::name:
```

Comment:
Just a question, may I request the same attribute multiple times?

Reply:
It means we can de-duplicate attributes before iteration (Including package IDs), but this is not required for this PR.